From 26371ccfe0728b8bdd3f8ae3109d22645e385525 Mon Sep 17 00:00:00 2001
From: Kjartan Maraas <kmaraas@gnome.org>
Date: Tue, 14 May 2013 13:53:52 +0200
Subject: [PATCH 14/14] Commit patch from bug #662623 to fix connection
 failures on heavily loaded systems. Patch from Red Hat support.

---
 linc2/src/linc-connection.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/linc2/src/linc-connection.c b/linc2/src/linc-connection.c
index 77bab0f..88169f9 100644
--- a/linc2/src/linc-connection.c
+++ b/linc2/src/linc-connection.c
@@ -651,8 +651,12 @@ link_connection_do_initiate (LinkConnection        *cnx,
 		fix_permissions (service);
 	}
 #endif
-
+retry:
 	LINK_TEMP_FAILURE_RETRY_SOCKET (connect (fd, saddr, saddr_len), rv);
+	if (rv == -1L && errno == EAGAIN) {
+		g_usleep (10000);
+		goto retry;
+	}
 #ifdef HAVE_WINSOCK2_H
 	if (rv == SOCKET_ERROR) {
 		if ((options & LINK_CONNECTION_NONBLOCKING) &&
-- 
1.8.3

